annotations_models/train_random_forest_classifier_model.ipynb

```diff
@@ -46,31 +46,85 @@
     "from sklearn.ensemble import RandomForestClassifier\n",
     "from sklearn.preprocessing import StandardScaler\n",
     "import pprint\n",
+    "import os\n",
     "\n",
-    "from training.train import train_imbalanced_model, cross_validation_balance_on_date\n",
+    "from training.train import train_imbalanced_model, cross_validation_balance_on_date, create_folds\n",
     "from training.utils import get_cross_validation_results_filepath, get_model_filepath\n",
-    "from training.metric_calculation import calculate_average_metrics, get_metrics"
+    "from training.metric_calculation import calculate_average_metrics, get_metrics\n",
+    "import requests\n",
+    "\n",
+    "data_dir = os.path.abspath(os.path.join(os.getcwd(), '..')).replace(\"\\\\\",\"/\")+\"/data/annotations/annotations_pixel_dataframes/\"\n",
+    "\n",
+    "\n",
+    "def download_file(url, local_path):\n",
+    "\n",
+    "    # Send a GET request to the URL to download the zip file\n",
+    "    response = requests.get(url)\n",
+    "\n",
+    "    # Check if the request was successful (status code 200)\n",
+    "    if response.status_code == 200:\n",
+    "        # Write the content of the response (the zip file) to a local file\n",
+    "        with open(local_path, \"wb\") as file:\n",
+    "            file.write(response.content)\n",
+    "\n",
+    "        print(f\"file has been downloaded to {local_path}\")\n",
+    "    else:\n",
+    "        print(\"Failed to download file:\", response.status_code)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 49,
    "metadata": {},
    "outputs": [],
    "source": [
     "# Set Variables\n",
-    "location = \"Coepelduynen\"\n",
-    "satellite_constellation = \"PNEO\"\n",
+    "location = \"Schippersgat\"\n",
+    "satellite_constellation = \"Superview\"\n",
     "\n",
     "if location == \"Voornes Duin\":\n",
-    "    annotated_pixels_filepath = \"C:/Users/pzhadmin/Data/remote-sensing/annotations/Voornes_Duin_PNEO_2024-01-29_pixel_annotations.parquet\"\n",
+    "\n",
```

Comment:
If I understand correctly most of this cell (except the first 2 lines) is to download files if they are not yet available locally, right? Could we make that into a separate function and place that in an I/O source python file? To me all the code feels a bit overwhelming when I look through this notebook and it only does a bit of convenient I/O.